admin_bans.php

```diff
@@ -76,6 +76,10 @@
 				$result = $db->query('SELECT registration_ip FROM '.$db->prefix.'users WHERE id='.$user_id) or error('Unable to fetch user info', __FILE__, __LINE__, $db->error());
 				$ban_ip = ($db->num_rows($result)) ? $db->result($result) : '';
 			}
+
+			// Disable localhost IP ban
+			if (in_array($ban_ip, array('', '127.0.0.1', '::1'), true))
+				$ban_ip = '';

```

Comment:
What I mean is this:

```php
if (in_array($ban_ip, array('', '127.0.0.1', '::1'), true))
```

This checks if `$ban_ip` is either `''`, `'127.0.0.1'` or `::1`.
If it is, you do this:

```php
$ban_ip = '';
```

Which is OK, but the first array item does not make sense anymore then, because it's the same as this:

```php
if ($ban_ip == '') {
    $ban_ip = '';
}
```

That's a pretty useless declaration. It already was `''` and you're setting it to be `''` once more. There's no need to change `$ban_ip` in case it's already `''`.